[docs] Add YSQL and YCQL client driver pages

stevebang opened the pull request:
Adds:
- new YSQL client drivers page
- new YCQL client drivers page
- updates naming in "Build an application" pages
- adds new Yugabyte Java Driver for YCQL (4.6.0-yb-x) to new listing of YCQL drivers
- updates "Build a Java application" for YCQL to use new 4.8 driver and code snippet using changed API (thanks, @nchandrappa !)
- update `wget` commands in examples that pointed to the dead central.maven.org, had `http` (no longer supported) instead of `https`, and old versions

nchandrappa commented:
@stevebang Spoke to @rkarthik007 regarding having sperate getting started examples for 3.x and 4.x version of YCQL java driver. Basically we are not recommending 4.x version for production workloads yet hence we want to keep both examples. Additionally, mark the 4.x version as not production-ready and it's a release candidate. Thanks

stevebang commented:
> @stevebang Spoke to @rkarthik007 regarding having sperate getting started examples for 3.x and 4.x version of YCQL java driver. Basically we are not recommending 4.x version for production workloads yet hence we want to keep both examples. Additionally, mark the 4.x version as not production-ready and it's a release candidate. Thanks

@nchandrappa I added a second "Build a Java application" button for "YCQL 4.6", reverted the original example to 3.8, and added a note that 4.6 is a "release candidate and no supported yet for production" to the "YCQL 4.6" application page and the listing in the YCQL client drivers listing for "Yugabyte Java Driver for YCQL 4.6". cc @rkarthik007
